web-common/src/features/canvas/components/charts/util.ts

```diff
@@ -29,3 +31,34 @@ export const chartMetadata: ChartMetadata[] = [
   { type: "bar_chart", title: "Bar", icon: BarChart },
   { type: "stacked_bar", title: "Stacked Bar", icon: StackedBar },
 ];
+
+export function mergedVlConfig(config: string): Config {
+  const defaultConfig = getRillTheme(true);
+  let parsedConfig: Config;
+
+  try {
+    parsedConfig = JSON.parse(config) as Config;
+  } catch {
+    console.warn("Invalid JSON config");
+    return defaultConfig;
+  }
+
+  const reverseArrayMerge = (
```

Comment:
Could you add a comment on why we need this reverse array merge?

Also since this is not really dependant on any function arguments, could you move this outside?

Reply:
This is required to put the values added by the user in the config before our default values. For example for category colors, we have our own set of 14 values. When users add their color, those would be put first.

I will add a comment and move this out to a separate file in a separate PR to avoid conflicts with PR forked against this.